### PR TITLE
Line shapes

### PR DESCRIPTION
**Pull request by EthanVanderHorn**
https://github.com/nanome-ai/nanome-core/pull/415

### CODE REVIEW

File: nanome/api/shapes/label.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
+from nanome._internal._shapes import _Label
+from nanome.util.enums import ShapeType
+from . import Shape
+
+
+class Label(_Label, Shape):
+    def __init__(self):
+        Shape.__init__(self, ShapeType.Label)
```
Comment (ajm13) [inline nanome/api/shapes/label.py:8]: doesn't `_Label.__init__` already call `_Shape.__init__`? same for the other shapes' inits

File: nanome/_internal/_shapes/_label.py [hunk 2]
```diff
@@ -0,0 +1,15 @@
+from ._anchor import _Anchor
+from ._shape import _Shape
+from nanome.util.enums import ShapeType
+
+
+class _Label(_Shape):
+    def __init__(self):
+        _Shape.__init__(self, ShapeType.Label)
+        self._anchors = [_Anchor._create(), _Anchor._create()]
```
Comment (ajm13) [inline nanome/_internal/_shapes/_label.py:9]: why does Label have 2 anchors by default?

File: nanome/api/shapes/label.py (hunk 1 above)
Comment (EthanVanderHorn) [reply nanome/api/shapes/label.py:8]: Yeah I was just following the pattern used by atom 🤔 I don't know what is necessary

File: nanome/_internal/_shapes/_serialization/_anchor_serializer.py [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+        return 0
+
+    def name(self):
+        return "ShapeAnchor"
+
+    def serialize(self, version, value, context):
+        context.write_long(value._target)
+        context.write_byte(int(value._anchor_type))
+        context.write_using_serializer(self._offset, value._local_offset)
+        context.write_using_serializer(self._offset, value._global_offset)
```
Comment (Astrovicis) [inline nanome/_internal/_shapes/_serialization/_anchor_serializer.py:19]: Is this capturing all three values all on self._offset? I'm confused

File: nanome/_internal/_shapes/_serialization/_anchor_serializer.py [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+        context.write_using_serializer(self._offset, value._viewer_offset)
+
+    def deserialize(self, version, context):
+        result = _Anchor._create()
+        result._target = context.read_long()
+        result._anchor_type = ShapeType.safe_cast(context.read_byte())
+        result._local_offset = context.read_using_serializer(self._offset)
+        result._global_offset = context.read_using_serializer(self._offset)
```
Comment (Astrovicis) [inline nanome/_internal/_shapes/_serialization/_anchor_serializer.py:27]: Same question here

**Review by ajm13 — APPROVED** [review]

File: nanome/_internal/_shapes/_serialization/_anchor_serializer.py (hunk 3 above)
Comment (ajm13) [reply nanome/_internal/_shapes/_serialization/_anchor_serializer.py:19]: `self._offset` is a UnityPositionSerializer, so yes

Comment (Astrovicis) [reply nanome/_internal/_shapes/_serialization/_anchor_serializer.py:19]: I totally forgot how plugin serialization worked for a sec..._offset is the serializer, not the shape property. my bad.

**Review by Astrovicis — APPROVED** [review]

**Pull request merged**